### DISCUSSION TITLE
build-rendered stylesheets are included twice

### DISCUSSION THREAD

**Discussion by richard-unterberg** [vikejs/vike#1811] [Discussion] [answered]
Hey guys, coming up with a odd one I discovered while using vike-react in ssr prod-mode with a little express setup.

### Steps to reproduce

- clone https://github.com/richard-unterberg/repro-vike-react-bug
- in root of cloned dir `npm install && npm run prod`
- open localhost:4248 
- Open Page Source Code

### Problem

In the outputted HTML there are two almost similiar named files referenced

![screen-1](https://github.com/user-attachments/assets/311c8a8f-8c66-45d0-abcb-8f8d876b02c0)

if we observe the files it seems they are also almost identical, with one slight addition to one of them

<img width="1828" alt="Bildschirmfoto 2024-08-16 um 23 24 16" src="https://github.com/user-attachments/assets/05d15bfd-c218-4612-9d71-6407a83cda4c">

This causes the browser to load both css files, which contains almost the same data

### Remarks

- If I remove the reference of the css file in `node_modules/vike-react/renderer/onRenderClient.js:6` and run `npm run prod` it will produce only one file, which is still weirdly named (can I change the name of this asset somehow?)
<img width="742" alt="Bildschirmfoto 2024-08-16 um 23 30 01" src="https://github.com/user-attachments/assets/2d714d33-e8ba-4986-a980-f8a9318ef531">

- tried to change from uno.css to tailwind with no change of the behaviour
- tried to exclude the `node_modules` folder in the uno config without noticing any change
- In the build log we can observe the generation of these two files happening in different build steps

<img width="662" alt="Bildschirmfoto 2024-08-16 um 23 35 29" src="https://github.com/user-attachments/assets/06740213-6d00-4af5-92a7-92aaf0f28747">

- After testing this some days now, I'm still not sure if this a vike-react problem or just a problem in my configuration..

Hope this helps. Thanks for the great work on the plugin 👍

**Comment by brillout** [comment]
Can you reproduce this with a minimal setup? In other words with `$ pnpm create bati --react` and without adding a CSS preprocessor?

I'm inclined to think that this isn't related to Vike, but let's see if we can find some indication that it is.

**Reply by richard-unterberg** [reply]
well, to my surprise yes, it also happens on the suggested bati code without touching anything than running `pnpm run build` and `pnpm run preview` - [here the repo](https://github.com/richard-unterberg/repro-vike-minimal-bug)

<img width="899" alt="Bildschirmfoto 2024-08-17 um 22 14 54" src="https://github.com/user-attachments/assets/7d06ec3e-e981-4ccc-8634-a71b57e29d9a">

also tested and happens with `pnpm create bati --react --tailwindcss --telefunc --express --eslint --prettier` too

**Reply by brillout** [reply]
Can you also reproduce with the following?
 - `pnpm create bati --react --tailwindcss`
 - `pnpm create bati --react`

**Reply by richard-unterberg** [reply]
yes, it happens on any bati installation which was suggested in this thread + the full blown one, I mentioned earlier the initial description.

- `pnpm create bati --react` (this one I tried already in my last answer) https://github.com/richard-unterberg/repro-vike-minimal-bug
- `pnpm create bati --react --tailwindcss`
- `pnpm create bati --react --tailwindcss --telefunc --express --eslint --prettier`

for further testing I was running the last installs with bati on a different devices (MB intel / MB M1) and tested npm and pnpm versios with the same result

I will check my "older" vike projects and will see what are the differences to them to further exclude that the issue is caused by my system.

**Reply by brillout** [reply]
I could reproduce indeed. Bug report: https://github.com/vikejs/vike/issues/1815.

**Reply by brillout** [reply, marked as answer]
Fixed in ~~`vike@0.4.186`~~ `vike@0.4.187`.

https://github.com/vikejs/vike/issues/1815#issuecomment-2295385978

**Reply by brillout** [reply]
Improved fix released in `vike@0.4.187`.

> can I change the name of this asset somehow?

The asset names are determined by Rollup, so I don't think you can.

The fix has a nice side effect that it improves the built CSS file names.

IIRC I asked you already but in case I didn't: we're looking for sponsors (#1350). It makes a huge difference to us!
